keras_contrib/layers/capsule.py

```diff
@@ -0,0 +1,211 @@
+       A normal pooling layer would lose a lot of
+       this information.
+
+       This layer can be used on the output of any layer
+       which has a 3-D output (including batch_size). For example,
+       in image classification, it can be used on the output of a
+       Conv2D layer for Computer Vision applications. Also,
+       it can be used on the output of a GRU or LSTM Layer
+       (Bidirectional or Unidirectional) for NLP applications.
+
+       The default activation function is 'linear'. But, this layer
+       is generally used with the 'squash' activation function
+       (recommended). To use the squash activation function, use :
+
+       def squash(x, axis=-1):
+           s_squared_norm = K.sum(K.square(x), axis, keepdims=True)
+                            + K.epsilon()
+           scale = K.sqrt(s_squared_norm) / (0.5 + s_squared_norm)
+           return scale * x
+
+       Then, create the capsule layer :
+
+       capsule = Capsule(num_capsule=10,
+                         dim_capsule=10,
+                         routings=3,
+                         share_weights=True,
+                         activation=squash)
+
+       # Example usage :
+           1). COMPUTER VISION
+
+           input_image = Input(shape=(None, None, 3))
+
+           conv_2d = Conv2D(64,
+                            (3, 3),
+                            activation='relu')(input_image)
+
+           capsule = Capsule(num_capsule=10,
+                             dim_capsule=16,
+                             routings=3,
+                             activation='relu',
+                             share_weights=True)(conv_2d)
+
+           2). NLP
+
+           maxlen = 72
+           max_features = 120000
+           input_text = Input(shape=(maxlen,))
+
+           embedding = Embedding(max_features,
+                                 embed_size,
+                                 weights=[embedding_matrix],
+                                 trainable=False)(input_text)
+
+           bi_gru = Bidirectional(GRU(64,
+                                      return_seqeunces=True))(embedding)
+
+           capsule = Capsule(num_capsule=5,
+                             dim_capsule=5,
+                             routings=4,
+                             activation='sigmoid',
+                             share_weights=True)(bi_gru)
+
+       # Arguments
+           num_capsule : Number of Capsules (int)
+           dim_capsules : Dimensions of the vector output of each Capsule (int)
+           routings : Number of dynamic routings in the Capsule Layer (int)
+           share_weights : Whether to share weights between Capsules or not
+           (boolean)
+           activation : Activation function for the Capsules (str)
+           regularizer : Regularizer for the weights of the Capsules (str)
+           initializer : Initializer for the weights of the Caspules (str)
+           cnstraint : Constraint for the weights of the Capsules (str)
```

Comment:
Typo. And note that it's not necessarily a string.

Reply:
Fixed the typo and removed 'str'. Included the option of adding initializers, regularizers and constraints without string.